index.rb

```diff
@@ -0,0 +1,75 @@
+module Enumerable
+  def my_each
+    i = 0
+    while self[i]
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    my_each { |e| yield(e, index(e)) }
+  end
+
+  def my_select
+    result = []
+    each do |e|
+      result << e if yield(e)
+    end
+    result
+  end
+
+  def my_all?
+    my_each do |e|
+      return false unless yield(e)
+    end
+    true
+  end
```

Comment:
- [ ] my_all? method should test for Regexp too. Check test cases [here](https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-all-3F)
- [ ] my_all? method should test for Classes too. Check test cases [here](https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-all-3F)
- [ ] my_all? method should return false when there's a false or nil in the array passed.